File: sdks/php/rest/src/Projects/Requests/CreateProjectRequest.php
<?php

namespace Brume\Rest\Projects\Requests;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

class CreateProjectRequest extends JsonSerializableType
{
    /**
     * @var string $name 1-128 chars.
     */
    #[JsonProperty('name')]
    public string $name;

    /**
     * @param array{
     *   name: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->name = $values['name'];
    }
}

File: sdks/csharp/rest/src/Brume.Rest/Projects/Requests/CreateProjectRequest.cs
using Brume.Rest.Core;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record CreateProjectRequest
{
    /// <summary>
    /// 1-128 chars.
    /// </summary>
    [JsonPropertyName("name")]
    public required string Name { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
